src/ai/defaultai.cc

```diff
@@ -194,10 +194,9 @@ DefaultAI::DefaultAI(Widelands::Game& ggame, Widelands::PlayerNumber const pid,
 			   break;
 
 		   case Widelands::NoteShip::Action::kWaitingForCommand:
-			   for (std::deque<ShipObserver>::iterator i = allships.begin(); i != allships.end();
-			        ++i) {
-				   if (i->ship == note.ship) {
-					   i->waiting_for_command_ = true;
+			   for (auto& observer : allships) {
```

Comment:
https://github.com/widelands/widelands/pull/4348#discussion_r503265977
I think you should follow the same advice here
```suggestion
			   for (ShipObserver& observer : allships) {
```